crates/red_knot_python_semantic/src/types/class.rs

```diff
@@ -769,7 +773,12 @@ impl<'db> ClassLiteralType<'db> {
                         continue;
                     }
 
-                    if class.is_known(db, KnownClass::Type) && policy.meta_class_no_type_fallback()
+                    // HACK: we should implement some more general logic here that supports arbitrary custom
+                    // metaclasses, not just `type` and `ABCMeta`.
+                    if matches!(
+                        class.known(db),
+                        Some(KnownClass::Type | KnownClass::ABCMeta)
+                    ) && policy.meta_class_no_type_fallback()
```

Comment:
I added this as a short-term hack to avoid the `__new__` false positives for classes that have `ABCMeta` as their metaclasses. Judging from the mypy_primer report, it looks extremely effective!